documentation/book/con-kafka-broker-exposing-external-listeners.adoc

```diff
@@ -0,0 +1,280 @@
+// Module included in the following assemblies:
+//
+// assembly-configuring-kafka-listeners.adoc
+
+[id='con-kafka-broker-exposing-external-listeners-{context}']
+= Exposing Kafka using {OpenShiftName} `Routes`
+
+An external listener of type `route` exposes Kafka by using {OpenShiftName} `Routes` and the HAProxy router.
+A dedicated `Route` is created for every Kafka broker pod.
+An additional `Route` is created to serve as a Kafka bootstrap address.
+Kafka clients can use these `Routes` to connect to Kafka on port 443.
+
+ifdef::Kubernetes[]
+NOTE: `Routes` are available only on {OpenShiftName}. External listeners of type `route` cannot be used on {KubernetesName}.
+endif::Kubernetes[]
+
+When exposing Kafka using {OpenShiftName} `Routes`, TLS encryption is always used.
+
+By default, the route hosts are automatically assigned by {OpenShiftName}.
+However, you can override the assigned route hosts by specifying the requested hosts in the `overrides` property.
+{ProductName} will not perform any validation that the requested hosts are available; you must ensure that they are free and can be used.
+
+.Example of an external listener of type `routes` configured with overrides for {OpenShiftName} route hosts
+[source,yaml,subs="attributes+"]
+----
+# ...
+listeners:
+  external:
+    type: route
+    authentication:
+      type: tls
+    overrides:
+      bootstrap:
+        host: bootstrap.myrouter.com
+      brokers:
+      - broker: 0
+        host: broker-0.myrouter.com
+      - broker: 1
+        host: broker-1.myrouter.com
+      - broker: 2
+        host: broker-2.myrouter.com
+# ...
+----
+
+For more information on using `Routes` to access Kafka, see xref:proc-accessing-kafka-using-routes-{context}[].
+
+= Exposing Kafka using loadbalancers
+
+External listeners of type `loadbalancer` expose Kafka by using `Loadbalancer` type `Services`.
+A new loadbalancer service is created for every Kafka broker pod.
+An additional loadbalancer is created to serve as a Kafka _bootstrap_ address.
+Loadbalancers listen to connections on port 9094.
+
+By default, TLS encryption is enabled.
+To disable it, set the `tls` field to `false`.
+
+For more information on using loadbalancers to access Kafka, see xref:proc-accessing-kafka-using-loadbalancers-{context}[].
+
```

Comment:
I think we should move the `.Example of an external listener of type `loadbalancer` using {KubernetesExternalDNS} annotations` up here. It's rather lost several pages down, and we don't have a xref here linking to it, so it's too easily missed.

Reply:
No, that should not be moved as that shows specifically the annotations. A new example should be added here without the annotations.